src/app_service/service/contacts/service.nim

```diff
@@ -355,36 +335,32 @@ QtObject:
 
     ## Returns contact details based on passed id (public key)
     ## If we don't have stored contact localy or in the db then we create it based on public key.
-    if(self.contacts.hasKey(pubkey)):
+    if self.contacts.hasKey(pubkey):
       return self.contacts[pubkey]
 
-    result = self.fetchContact(pubkey)
-    if result.dto.id.len == 0:
-      if(not pubkey.startsWith("0x")):
-        debug "id is not in a hex format"
-        return
+    if not pubkey.startsWith("0x"):
+      debug "id is not in a hex format"
+      return
 
-      var num64: int64
-      let parsedChars = parseHex(pubkey, num64)
-      if(parsedChars != PK_LENGTH_0X_INCLUDED):
-        debug "id doesn't have expected length"
-        return
+    var num64: int64
+    let parsedChars = parseHex(pubkey, num64)
+    if parsedChars != PK_LENGTH_0X_INCLUDED:
+      debug "id doesn't have expected length"
+      return
 
-      let alias = self.generateAlias(pubkey)
-      let trustStatus = self.getTrustStatus(pubkey)
-      let contact = self.constructContactDetails(
-        ContactsDto(
-          id: pubkey,
-          alias: alias,
-          ensVerified: result.dto.ensVerified,
-          added: result.dto.added,
-          blocked: result.dto.blocked,
-          hasAddedUs: result.dto.hasAddedUs,
-          trustStatus: trustStatus
-        )
+    let contact = self.constructContactDetails(
```

Comment:
I am wondering how this will impact the app. What's the scenario for contact to be default constructed like this?

Reply:
When we do not know about the contact at all. So someone that never shared their profile or we never fetched it.
This happens a lot in the Status community where people join but never post anything.

The old code would check the cache, then ask status-go, aka the status-go cache, but 100% of the time, it would also fail to get the contact because the status-go cache and the nim cache end up containing the same contacts anyway, especially on app start.

During the runtime of the app, the nim cache gets updated when we receive new contacts through signals.